diff --git a/services/kawaigazprombank/contracts/card/Card.sol b/services/kawaigazprombank/contracts/card/Card.sol
index 822f3dd..6017581 100644
--- a/services/kawaigazprombank/contracts/card/Card.sol
+++ b/services/kawaigazprombank/contracts/card/Card.sol
@@ -123,19 +123,6 @@ contract Card is ICard {
         uint256 to_id,
         uint256 sign_id
     ) external {
-        assembly {
-            mstore(0x40, 0xf0)
-        }
-
-        assembly {
-            mstore(0x00, from_id)
-            mstore(0x20, 5)
-            let k := keccak256(0x00, 0x40)
-            mstore(0x00, gift_id)
-            mstore(0x20, k)
-            mstore(0x80, sload(add(keccak256(0x00, 0x40), 2)))
-        }
-
         assembly {
             mstore(0x00, to_id)
             mstore(0x20, 1)
@@ -181,8 +168,8 @@ contract Card is ICard {
         bytes32 name
     ) public pure returns (bytes32 res) {
         uint256 to = uint256(uint160(gift.to));
+        bytes32 key = gift.key;
         assembly {
-            let key := mload(0x80)
             mstore(0x00, blk)
             res := xor(key, xor(keccak256(0x00, 0x20), xor(name, to)))
         }
