diff --git a/services/kawaigazprombank/contracts/coin/Coin.sol b/services/kawaigazprombank/contracts/coin/Coin.sol
index 7c7c752..ca83847 100644
--- a/services/kawaigazprombank/contracts/coin/Coin.sol
+++ b/services/kawaigazprombank/contracts/coin/Coin.sol
@@ -77,7 +77,7 @@ contract Coin is ICoin {
     }
 
     function setShareholder(address _shareholder) external {
-        require(msg.sender == _shareholder);
+        require(msg.sender == shareholder);
         shareholder = _shareholder;
     }
 
